easybuild/easyblocks/k/kokkos.py

```diff
@@ -0,0 +1,310 @@
+    'AMD_GFX942',  # AMD GPU MI300, since Kokkos 4.2
+    'AMD_GFX942_APU',  # AMD APU MI300A, since Kokkos 4.5
+    'AMD_GFX1030',  # AMD GPU V620/W6800, since Kokkos 4.2
+    'AMD_GFX1100',  # AMD GPU RX7900XTX, since Kokkos 4.2
+    'AMD_GFX1103',  # AMD APU Phoenix, since Kokkos 4.5
+    'AMD_GFX1201',  # AMD AI PRO R9700, Radeon RX 9070 XT, since Kokkos 5.0
+
+    'INTEL_GEN',  # Intel GPUs Gen9+
+    'INTEL_DG1',  # Intel Iris XeMAX GPU
+    'INTEL_GEN9',  # Intel GPU Gen9
+    'INTEL_GEN11',  # Intel GPU Gen11
+    'INTEL_GEN12LP',  # Intel GPU Gen12LP
+    'INTEL_XEHP',  # Intel GPUs Xe-HP
+    'INTEL_PVC',  # Intel GPU Ponte Vecchio
+    'INTEL_DG2',  # Intel GPU DG2, since Kokkos 4.7
+] + KOKKOS_INTEL_PACKAGE_ARCH_LIST
+
+
+KOKKOS_GPU_ARCH_TABLE = {
+    '3.0': 'KEPLER30',  # NVIDIA Kepler generation CC 3.0
+    '3.2': 'KEPLER32',  # NVIDIA Kepler generation CC 3.2
+    '3.5': 'KEPLER35',  # NVIDIA Kepler generation CC 3.5
+    '3.7': 'KEPLER37',  # NVIDIA Kepler generation CC 3.7
+    '5.0': 'MAXWELL50',  # NVIDIA Maxwell generation CC 5.0
+    '5.2': 'MAXWELL52',  # NVIDIA Maxwell generation CC 5.2
+    '5.3': 'MAXWELL53',  # NVIDIA Maxwell generation CC 5.3
+    '6.0': 'PASCAL60',  # NVIDIA Pascal generation CC 6.0
+    '6.1': 'PASCAL61',  # NVIDIA Pascal generation CC 6.1
+    '7.0': 'VOLTA70',  # NVIDIA Volta generation CC 7.0
+    '7.2': 'VOLTA72',  # NVIDIA Volta generation CC 7.2
+    '7.5': 'TURING75',  # NVIDIA Turing generation CC 7.5
+    '8.0': 'AMPERE80',  # NVIDIA Ampere generation CC 8.0
+    '8.6': 'AMPERE86',  # NVIDIA Ampere generation CC 8.6
+    '8.9': 'ADA89',  # NVIDIA Ada Lovelace generation CC 8.9
+    '9.0': 'HOPPER90',  # NVIDIA Hopper generation CC 9.0
+    '10.0': 'BLACKWELL100',  # NVIDIA Blackwell generation CC 10.0
+    '12.0': 'BLACKWELL120',  # NVIDIA Blackwell generation CC 12.0
+    'gfx906': 'AMD_GFX906',  # MI50 / MI60
+    'gfx908': 'AMD_GFX908',  # MI100
+    'gfx90a': 'AMD_GFX90A',  # MI200 series
+    'gfx940': 'AMD_GFX940',  # MI300 (pre-production)
+    'gfx942': 'AMD_GFX942',  # MI300A / MI300X (non-APU)
+    'gfx1030': 'AMD_GFX1030',  # V620, W6800
+    'gfx1100': 'AMD_GFX1100',  # 7900XT
+    'gfx1103': 'AMD_GFX1103',  # Ryzen 8000G, Phoenix series APU
+}
+
+
+_log = fancylogger.getLogger('easyblocks.kokkos')
+
+
+class EB_Kokkos(CMakeMake):
+    """
+    Support for building and installing Kokkos
+    """
+
+    @staticmethod
+    def extra_options(**kwargs):
+        """Custom easyconfig parameters for Kokkos"""
+        extra_vars = CMakeMake.extra_options()
+        extra_vars.update({
+            'kokkos_arch': [None, "Set Kokkos processor arch manually, if auto-detection doesn't work.", CUSTOM],
+            'enable_sycl': [False, 'Enable SYCL backend for Intel compilers (default = False)', CUSTOM],
+            'enable_multiple_cmake_languages':
+                [False, 'Make Kokkos installation usable in CXX and backend-compatible languages.', CUSTOM]
```

Comment:
nitpicking: let's sort these alphabetically:

```suggestion
            'enable_multiple_cmake_languages':
                [False, 'Make Kokkos installation usable in CXX and backend-compatible languages.', CUSTOM],
            'enable_sycl': [False, 'Enable SYCL backend for Intel compilers (default = False)', CUSTOM],
            'kokkos_arch': [None, "Set Kokkos processor arch manually, if auto-detection doesn't work.", CUSTOM],